content/apply.js

```diff
@@ -316,78 +203,61 @@ const APPLY = (() => {
     }).catch(console.error);
   }
 
-  function rootReady() {
+  // using a callback to run it immediately
+  // unlike Promises which resolve at the end of event loop cycle
+  function rootReady(cb) {
     if (document.documentElement) {
-      return Promise.resolve();
-    }
-    return new Promise(resolve => {
+      cb();
```

Comment:
Actually, a site's script may easily force synchronous layout between this moment and the time the Promise resolves so no, it's not a microoptmization, but an important thing. I guess I'll add a comment there.

Reply:
> the callback the code is actually more consistent because evade also uses a callback.

It becomes inconsistent with other `-ready` functions. I don't think it can be compared to `evade` because `evade` needs a callback.

> a site's script may easily force synchronous layout between this moment and the time the Promise resolves

If a page script decides to do this, it has a higher chance to happen during `getSectionsByUrl` instead of between two microtasks and there is no way to prevent it.

I'm not against micro-optimization. I just want to point out why I think it is micro.

Reply:
Nope, getSectionsByUrl will usually return the result before any of the site scripts run in 99.9% of cases (only the trivial and small inline scripts in <head> might run before that).

Reply:
Actually it's possible to make it return the result even before the first script in head runs :-) by using chrome.declarativeContent API which injects earlier than document_start. I just didn't want to add it into this PR.

Reply:
Found a better way: rootReady is now an internal thing in root observer.